### PR TITLE
Add ElectionConfiguration struct to services config

### PR DESCRIPTION
**Pull request by jskelcy**
This seemed to be a general config that should live in the repo.

### CODE REVIEW

File: services/config.go [hunk 1]
```diff
@@ -59,3 +61,18 @@ func (cfg Configuration) NewOptions() Options {
 	}
 	return opts
 }
+
+// ElectionConfiguration is for configuring election timeouts and TTLs
+type ElectionConfiguration struct {
+	LeaderTimeout time.Duration `yaml:"leaderTimeout"`
```
Comment (cw9) [inline services/config.go]: Can you make them all optional fields? like [this](https://github.com/m3db/m3msg/blob/master/producer/config/buffer.go)

File: services/config.go [hunk 2]
```diff
@@ -59,3 +61,26 @@ func (cfg Configuration) NewOptions() Options {
...
+	if cfg.LeaderTimeout != nil {
+		opts = opts.SetTTLSecs(*cfg.TTLSeconds)
+	}
+	if cfg.ResignTimeout != nil {
+		opts = opts.SetLeaderTimeout(*cfg.LeaderTimeout)
+	}
+	if cfg.ResignTimeout != nil {
+		opts = opts.SetResignTimeout(*cfg.ResignTimeout)
+	}
+
```
Comment (cw9) [inline services/config.go]: nit: remove empty line

File: services/config.go [hunk 3]
```diff
@@ -59,3 +61,26 @@ func (cfg Configuration) NewOptions() Options {
...
+type ElectionConfiguration struct {
+	LeaderTimeout *time.Duration `yaml:"leaderTimeout"`
+	ResignTimeout *time.Duration `yaml:"resignTimeout"`
+	TTLSeconds    *int           `yaml:"TTLSeconds"`
+}
+
+// NewOptions creates an ElectionOptions.
+func (cfg ElectionConfiguration) NewOptions() ElectionOptions {
+	opts := NewElectionOptions()
+	if cfg.LeaderTimeout != nil {
```
Comment (xichen2020) [inline services/config.go]: This should be `TTLSeconds`.

File: services/config.go [hunk 4]
```diff
@@ -59,3 +61,26 @@ func (cfg Configuration) NewOptions() Options {
...
+		opts = opts.SetTTLSecs(*cfg.TTLSeconds)
+	}
+	if cfg.ResignTimeout != nil {
```
Comment (xichen2020) [inline services/config.go]: This should be `LeaderTimeout`.

File: services/config.go [hunk 5]
```diff
@@ -59,3 +61,26 @@ func (cfg Configuration) NewOptions() Options {
...
+// ElectionConfiguration is for configuring election timeouts and TTLs
+type ElectionConfiguration struct {
+	LeaderTimeout *time.Duration `yaml:"leaderTimeout"`
+	ResignTimeout *time.Duration `yaml:"resignTimeout"`
+	TTLSeconds    *int           `yaml:"TTLSeconds"`
+}
+
+// NewOptions creates an ElectionOptions.
+func (cfg ElectionConfiguration) NewOptions() ElectionOptions {
+	opts := NewElectionOptions()
```
Comment (xichen2020) [inline services/config.go]: I also prefer to order the setters in the same way as the fields in the config struct.

**Comment by cw9** [comment]
How did the test pass? Can you make sure the test catches those problems?

**Pull request merged**